src/stats.rs

```diff
@@ -76,6 +75,53 @@ pub struct Stats {
 }
 
 impl Stats {
+    // Create a new instance, with the given number of routes
+    pub fn new(num_routes: usize) -> Self {
+        Stats {
```

Comment:
Um, problem with using the `..Default::default()` notation is that it requires `Stats` to support `Default` which means the user can simply forget to specify the number of routes, causing breakage. So I'd prefer leaving this explicit actually.